### DISCUSSION TITLE
Steady-state simulation using THM

### DISCUSSION THREAD

**Discussion by NorthMagic** [idaholab/moose#27781] [Q&A Modules: Thermal Hydraulics] [answered]
Hello, I encountered a problem at run time with "Vector tag 'TIME' does not exist. This may occur if you have a TimeKernel in your problem but did not specify a transient executioner". When I simulated a single flow channel without junctions, THM using the Steady executioner worked without error. But for a singlel flow channel with junctions, THM using the Steady executioner failed to run.

Is it unsupported to run THM with the Steady executioner for a model including junctions? Do I have to simulate this steady-state problem using the Transient executioner by setting `steady_state_dection=true`?

**Comment by GiudGiud** [comment, marked as answer]
Hello

A junction you are using must have a time kernel. 

Either:

Find which junction that is and add code to prevent adding the time kernel in a steady solve

Use a transient executioner and let it run to steady state. It's the same as a steady solve in terms of results

**Reply by GiudGiud** [reply]
@joshuahansel there might be a thing to change in a junction here

**Reply by NorthMagic** [reply]
Thank you very much for your rapid reply! I think the first choice you given is better than the second. Using a transient executioner and let it run to steady state can get identical results, but the computing time will be longer than the Steady executioner?

**Reply by NorthMagic** [reply]
> @joshuahansel there might be a thing to change in a junction here

I am using `JunctionParallelChannels1Phase`, from which `ODETimeDerivative` is added by `addMooseObjects` as shown in its source file. From my point, a small change to this routine is gonna solve my problem, is it right?

**Reply by GiudGiud** [reply]
I think so

**Reply by NorthMagic** [reply]
I made a small change to `addMooseObjects`,  just let it omit the addition of `ODETimeDerivative`. For a single flow channel with junctions by using the `Steady` executioner, the mentioned problem was solved. But I encountered a new problem. THM got an error with the following:
 0 Nonlinear |R| = 2.295593e+02
  Linear solve did not converge due to DIVERGED_PC_FAILED iterations 0
                 PC failed due to FACTOR_NUMERIC_ZEROPIVOT
Nonlinear solve did not converge due to DIVERGED_FNORM_NAN iterations 0
 Solve Did NOT Converge!
Aborting as solve did not converge

I don't know what problem it is. Could you please give me some advice?

**Reply by GiudGiud** [reply]
Can you try adding -pc_factor_shift NONZERO to your petsc options?

There is a troubleshooting page on our website for convergence issues

**Reply by joshuahansel** [reply]
Also, maybe it could be your initial guess? I know I've tried Steady simulations with THM before and diverged with a bad initial guess, like the velocity being zero.

**Reply by NorthMagic** [reply]
This unconverged problem was solved by using improved petsc options, which are recommended by the Porous_Flow module. Here is the url: https://mooseframework.inl.gov/modules/porous_flow/solvers.html. Thank you two very much!
